### PR TITLE
Make conditional discard flags available externally

### PR DESCRIPTION
**Pull request by kuhar**
This is so that we can set/access these flags in other `.cpp` files.

### CODE REVIEW

File: llvm/lib/Target/AMDGPU/AMDGPUConditionalDiscard.cpp [hunk 1]
```diff
@@ -70,19 +70,25 @@
 using namespace llvm;
 using namespace llvm::AMDGPU;
 
+namespace llvm {
+namespace cl {
+
 // Enable conditional discard transformations
-static cl::opt<bool> EnableConditionalDiscardTransformations(
+opt<bool> EnableConditionalDiscardTransformations(
```
Comment (jayfoad) [inline llvm/lib/Target/AMDGPU/AMDGPUConditionalDiscard.cpp:77]: No particular objection, but I haven't seen this pattern before of putting the option itself (`EnableConditionalDiscardTransformations`) in the `cl` namespace. Is it used elsewhere in LLVM?

Comment (perlfu) [reply llvm/lib/Target/AMDGPU/AMDGPUConditionalDiscard.cpp:77]: It does seem to be used occasionally elsewhere in LLVM.
Exposing this option seems fine to me.

Comment (kuhar) [reply llvm/lib/Target/AMDGPU/AMDGPUConditionalDiscard.cpp:77]: I've seen this pattern used in LLPC, e.g.,
- https://github.com/GPUOpen-Drivers/llpc/blob/dev/llpc/context/llpcCompiler.cpp#L105
- https://github.com/GPUOpen-Drivers/llpc/blob/dev/llpc/util/llpcDebug.cpp#L47
- https://github.com/GPUOpen-Drivers/llpc/blob/dev/llpc/util/llpcTimerProfiler.cpp#L49

Comment (kuhar) [reply llvm/lib/Target/AMDGPU/AMDGPUConditionalDiscard.cpp:77]: We could put it in namespace llvm instead of llvm::cl. Both work for me.

**Pull request closed without merging**